src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -50,6 +51,33 @@ FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::~FFN()
       boost::apply_visitor(deleteVisitor));
 }
 
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+void FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::CheckInputShape(
```

Comment:
Hey @NippunSharma thanks for contributing.
I think if you add this in `util` it will be more better. What do you think? @zoq

Reply:
You mean an `util` folder in the ann folder, I think that would make sense.

Reply:
@zoq , @himanshupathak21061998 I think this would be a good idea. I will add the `util` folder inside `ann` and put the `CheckInputShape` function there.